### ISSUE TITLE
(bio)schema.org markup for Tools listed on dockstore.org

### ISSUE THREAD

**Issue by mr-c** [ga4gh/dockstore#604]
## Feature Request 

(bio)schema.org markup for Tools listed on dockstore.org

Specification is at https://docs.google.com/document/d/1fQUPkjNnHfbx3dHtPeV_QXy_-DhUypoXliX1jE0AaYc/edit

### Desired behaviour

Visiting https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fdockstore.org%2Fcontainers%2Fquay.io%2Fmr_c%2Fkhmer
would list the structured data extracted from the `khmer` tool entry

**Comment by denis-yuen** [comment]
I think the second link is to 
https://dockstore.org/containers/quay.io/mr_c/khmer
which leads to 
https://github.com/mr-c/khmer

I can't find a CWL file though, is it on a particular branch?

**Comment by mr-c** [comment]
@denis-yuen Yeah, that is a rather old entry, something more recent would probably be a better test case

https://github.com/mr-c/khmer/tree/docker-2.0 is the branch

**Comment by denis-yuen** [comment]
Hmm, I can find the Dockerfile https://github.com/mr-c/khmer/blob/docker-2.0/docker/Dockerfile
Can't find the CWL file though

**Comment by mr-c** [comment]
@denis-yuen There isn't one yet, this was from an early test I did of dockstore.org to get a sense of its capabilities

Lets use https://dockstore.org/containers/quay.io/ljdursi/pcawg-merge-annotate & https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fdockstore.org%2Fcontainers%2Fquay.io%2Fljdursi%2Fpcawg-merge-annotate as the test case

**Comment by denis-yuen** [comment]
@k-cao  FYI, will explain further after the demo. https://developers.google.com/search/docs/guides/intro-structured-data and https://github.com/ga4gh/dockstore/issues/341 are also related

**Comment by denis-yuen** [comment]
@k-cao need to review on staging.dockstore.org

**Comment by denis-yuen** [comment]
Doesn't seem to work yet

**Comment by mr-c** [comment]
Ping :-)

**Comment by denis-yuen** [comment]
Unfortunately, I think no progress was made on this thread. 
IIRC, the sticking point was finding a way to pass along the structured data (or any other data useful for google search in #341) to something that is not javascript aware. I know in the "old-days" of AngularJS, there were pre-rendering solutions like https://prerender.io/ but I haven't looked for the Angular (2) equivalent yet.

**Comment by mr-c** [comment]
@denis-yuen Is the whole page rendered client side?

Also, see this: https://twitter.com/eocarragain/status/940220166201372672

**Comment by denis-yuen** [comment]
Yup. looking briefly, that doesn't seem to necessarily be a barrier to that structured data testing tool though. We just need to carve out time to experiment.

**Comment by cricketsloan** [comment]
[~rjbautis] and [~sradesh] 

read up on this
